### ISSUE TITLE
[MINOR FEATURE] Improve the `AutomaticCommunicationalPattern` Knowledge Source 

### ISSUE THREAD

**Issue by PSchmiedmayer** [Apodini/Apodini#318]
**Problem**
Currently a `Handler` like is detected as using the `.bidirectionalStream` by the `AutomaticCommunicationalPattern` based on the fact that it returns a `Response<...>` response of the `handle` function even though it might just use the `Response<...>` as a way to return additional `Information` to the client.

**Reproducible Example**
The following `Handler` is detected as `.bidirectionalStream`:
```swift
struct BlobGreeter: Handler {
    @Parameter(.http(.path)) var name: String
    
    @Parameter(.http(.query)) var greeting: String?
    
    
    func handle() -> Apodini.Response<Blob> {
        Response.send(
            Blob(Data("\(greeting ?? "Hello"), \(name)!".utf8), type: .text(.html)),
            information: [AnyHTTPInformation(key: "Test", rawValue: "Test")]
        )
    }
}
```
You can let Apodini know about the correct pattern using the
```swift
var metadata: Metadata {
    Pattern(.requestResponse)
}
```
metadata.

**Proposed Solution**
The expected behavior would be to classify the Handler shown above as a `.requestResponse` pattern as it does not include a `@State`, `@ObservedObject`, or `@Environment` value that could be observed.
The `AutomaticCommunicationalPattern` should be extended by also checking the available properties in the the Handler and any delegating handlers it uses to detect if `@State`, `@ObservedObject`, or `@Environment` (using an observable object) could be used to trigger a web service side streaming event.

**Comment by theMomax** [comment]
I would consider this more of a missing feature than a bug. The problem is that even if an endpoint uses no `@State`, `AnyObservedObject` and does not inspect `@Environment(\.connection)` (giving it access to the `ConnectionState`), it could still be intended to be a bidirectional streaming endpoint as long as it returns a `Response`. This endpoint would violate all our design guidelines, but it is possible. The intention behind `AutomaticCommunicationalPattern` was to only restrict the pattern if it is **certain**. I.e. when writing an `InterfaceExporter` and the `AutomaticCommunicationPattern` is `.requestResponse`, I can safely assume the first evaluation of the `Handler` will result in a `Response` with `ConnectionEffect.close`.

I see two possible options:

* We change the strategy behind the automatic detection to be a _best guess_ (even if that is not certain) and use the heuristics you suggested. In that case we'd have to think about how this affects handling of evaluation in exporters.
* We refactor the `Response` type system to separate the concerns of carrying `Information` and defining `ConnectionEffect` so we can make a decision with certainty.

**Comment by PSchmiedmayer** [comment]
@theMomax Thanks for your input! 
I think it makes most sense to go for a "best guess" approach as detailed in your first option. This would provides us some good results for most of the cases and then allows a developer to still override the automatic best guess with a manual metadata if needed. Nevertheless this would in summary reduce the amount of "misclassifications" by a lot and probably even catches a large fraction of cases encountered in the wild as long as they adhere to the guidelines defined by Apodini.

**Comment by theMomax** [comment]
Fixed with #346
